### ISSUE TITLE
Using perform_at method the job doesn't get enqueued

### ISSUE THREAD

**Issue by tgmerritt** [mperham/sidekiq#4116]
Ruby version:
Sidekiq / Pro / Enterprise version(s):  Regular Sidekiq

Please include your initializer and any error message with the full backtrace.

Initializer:

```
require 'sidekiq'
require 'sidekiq/web'

Sidekiq::Web.use(Rack::Auth::Basic) do |user, password|
  [user, password] == [ENV['SIDEKIQ_USERNAME'], ENV['SIDEKIQ_PASSWORD']]
end

if Rails.env.development?
  if ENV['SIDEKIQ_INLINE'] == 'false'
      puts '=========================================='
      puts '| Using real sidekiq worker (not inline) |'
      puts '=========================================='
  else
      puts '==============================='
      puts '| Using inline sidekiq worker |'
      puts '==============================='
    require 'sidekiq/testing'
    Sidekiq::Testing.inline!
  end
end
```

Are you using an old version?

Version:

```
2019-03-02T14:38:34.135Z 7485 TID-owyi8wmmh INFO: Running in ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-darwin18]
2019-03-02T14:38:34.135Z 7485 TID-owyi8wmmh INFO: See LICENSE and the LGPL-3.0 for licensing details.
2019-03-02T14:38:34.135Z 7485 TID-owyi8wmmh INFO: Upgrade to Sidekiq Pro for more features and support: http://sidekiq.org
2019-03-02T14:38:34.135Z 7485 TID-owyi8wmmh INFO: Booting **Sidekiq 5.2.3** with redis options {:id=>"Sidekiq-server-PID-7485", :url=>nil}
2019-03-02T14:38:34.136Z 7485 TID-owyi8wmmh INFO: Starting processing, hit Ctrl-C to stop
```

Have you checked the changelogs to see if your issue has been fixed in a later version?

I checked - there is no mention of this issue

https://github.com/mperham/sidekiq/blob/master/Changes.md
https://github.com/mperham/sidekiq/blob/master/Pro-Changes.md
https://github.com/mperham/sidekiq/blob/master/Ent-Changes.md

Issue:

```
[18] pry(main)> BroadcastMessageJob.perform_at(@announcement.send_at+7.hours, @announcement.id)
  Announcement Load (0.3ms)  SELECT  "announcements".* FROM "announcements" WHERE "announcements"."id" = $1 LIMIT 1  [["id", 4]]
  User Load (0.4ms)  SELECT "users".* FROM "users" WHERE (usertype IN (2) AND status = 1)
"442f57e145ab7b23b01e55e2"
[20] pry(main)> Sidekiq::Queue.new.find_job("442f57e145ab7b23b01e55e2")
nil

```

BroadcastMessageJob:

```
class BroadcastMessageJob
  include Sidekiq::Worker

  def perform(announcement_id)
    @announcement = Announcement.find announcement_id

    query = []
    query << User::TYPE_TECH if @announcement.technicians
    query << User::TYPE_MANAGER if @announcement.managers
    query << User::TYPE_ADMIN if @announcement.admins
      
    if @announcement.all_users
      @users = User.where('usertype IN (?) AND status = ?', [User::TYPE_ADMIN, User::TYPE_MANAGER, User::TYPE_DISPATCHER, User::TYPE_TECH], User::STATUS_ACTIVE)
    else 
      @users = User.where('usertype IN (?) AND status = ?', query, User::STATUS_ACTIVE)
    end

    @users.each do |u|
      AdminMailer.announcement_via_text(@announcement.id, u) if @announcement.message_type == "SMS"
      AdminMailer.announcement_via_email(@announcement.id, u) if @announcement.message_type == "Email"
    end
  end
end
```

Announcement Object used in the test above:

```
#<Announcement:0x00007ffd853ed048> {
              :id => 4,
            :text => "Something",
    :message_type => "Email",
         :user_id => 15,
     :technicians => false,
        :managers => false,
          :admins => true,
       :all_users => false,
           :delay => nil,
         :send_at => Sat, 02 Mar 2019 08:30:00 UTC +00:00,
      :created_at => Sat, 02 Mar 2019 14:31:26 UTC +00:00,
      :updated_at => Sat, 02 Mar 2019 14:31:26 UTC +00:00
}
```

send_at:
```
@announcement.send_at.class
ActiveSupport::TimeWithZone < Object
```

Responds to to_f (like you mention here https://github.com/mperham/sidekiq/issues/1689 ):

```
@announcement.send_at.to_f
1551515400.0
```

Bottom line: **I get a Job ID after calling the perform_at method but the Job doesn't actually get into Sidekiq - why not?**

**Comment by mperham** [comment]
It’s not enqueued yet, it’s in the scheduled set. See the API wiki page.
